lms/static/js/spec/search/search_spec.js

```diff
@@ -285,145 +194,372 @@ define([
+            beforeEach(function () {
+                beforeEachHelper.call(this, CourseSearchItemView);
+            });
+            it('renders items correctly', rendersItem);
+            it('renders Sequence items correctly', rendersSequentialItem);
+            it('logs view event', logsSearchItemViewEvent);
         });
 
-        it('shows loading message', function () {
-            this.listView.showLoadingMessage();
-            expect($('#course-content')).toBeHidden();
-            expect(this.listView.$el).toBeVisible();
-            expect(this.listView.$el).not.toBeEmpty();
+        describe('DashSearchItemView', function () {
+            beforeEach(function () {
+                beforeEachHelper.call(this, DashSearchItemView);
+            });
+            it('renders items correctly', rendersItem);
+            it('renders Sequence items correctly', rendersSequentialItem);
+            it('displays course name in breadcrumbs', function () {
+                expect(this.seqItem.$el.find('.sri-course-name')).toContainHtml(this.model.get('course_name'));
+            });
+            it('logs view event', logsSearchItemViewEvent);
         });
 
-        it('shows error message', function () {
-            this.listView.showErrorMessage();
-            expect($('#course-content')).toBeHidden();
-            expect(this.listView.$el).toBeVisible();
-            expect(this.listView.$el).not.toBeEmpty();
+    });
+
+
+    describe('SearchForm', function () {
+
+        function trimsInputString() {
+            var term = '    search string  ';
+            $('.search-field').val(term);
+            $('form').trigger('submit');
+            expect(this.onSearch).toHaveBeenCalledWith($.trim(term));
+        }
+
+        function doesSearch() {
+            var term = '  search string  ';
+            $('.search-field').val(term);
+            this.form.doSearch(term);
+            expect(this.onSearch).toHaveBeenCalledWith($.trim(term));
+            expect($('.search-field').val()).toEqual(term);
+            expect($('.search-field')).toHaveClass('is-active');
+            expect($('.search-button')).toBeHidden();
+            expect($('.cancel-button')).toBeVisible();
+        }
+
+        function triggersSearchEvent() {
+            var term = 'search string';
+            $('.search-field').val(term);
+            $('form').trigger('submit');
+            expect(this.onSearch).toHaveBeenCalledWith(term);
+        }
+
+        function changesToActiveState() {
+            var term = 'search string';
+            $('.search-field').val(term);
+            $('form').trigger('submit');
+            expect($('.search-field')).toHaveClass('is-active');
+            expect($('.search-button')).toBeHidden();
+            expect($('.cancel-button')).toBeVisible();
+        }
+
+        function clearsSearchOnCancel() {
+            $('.search-field').val('search string');
+            $('.cancel-button').trigger('click');
```

Comment:
Why would this cancel do anything. Search hasn't been triggered before canceling..